example(customer_thermometer):Add customer thermometer connector example (#243)
fivetran/fivetran_connector_sdk: examples/source_examples/customer_thermometer/README.md

# Customer Thermometer API Connector Example

## Connector overview

> The Customer Thermometer connector for Fivetran fetches customer feedback data from the Customer Thermometer API and syncs it to your data warehouse. This connector supports multiple endpoints including comments, blast results, recipient lists, thermometers, and feedback metrics. The connector implements API key authentication, parses XML responses, and is stateless, following Fivetran best practices for reliability, security, and maintainability.

Reviewer: ```suggestion
The [Customer Thermometer](https://www.customerthermometer.com/) custom connector for Fivetran fetches customer feedback data from the Customer Thermometer API and syncs it to your destination. This connector supports multiple endpoints including comments, blast results, recipient lists, thermometers, and feedback metrics. 

The connector implements API key authentication, parses XML responses, and is stateless, following Fivetran best practices for reliability, security, and maintainability.
```

## Requirements

- [Supported Python versions](https://github.com/fivetran/fivetran_connector_sdk/blob/main/README.md#requirements)
- Operating system:
  - Windows: 10 or later (64-bit only)
  - macOS: 13 (Ventura) or later (Apple Silicon [arm64] or Intel [x86_64])
  - Linux: Distributions such as Ubuntu 20.04 or later, Debian 10 or later, or Amazon Linux 2 or later (arm64 or x86_64)

## Getting started

Refer to the [Connector SDK Setup Guide](https://fivetran.com/docs/connectors/connector-sdk/setup-guide) to get started.

## Features

The connector supports the following features:

> - Multiple endpoints: Supports comments, blast results, recipient lists, thermometers, and metrics
> - API key authentication: Secure authentication using Customer Thermometer API keys
> - XML parsing: Converts XML API responses to structured records
> 
> - Error handling: Individual record and API error handling without stopping entire sync
> - Comprehensive logging: Uses Fivetran's logging framework for troubleshooting and monitoring
> - SDK v2.0.0+ compliance: Uses direct operation calls (no yield statements)

Reviewer: ```suggestion
- Multiple endpoints: Supports comments, blast results, recipient lists, thermometers, and metrics
- API key authentication: Secure authentication using Customer Thermometer API keys
- XML parsing: Converts XML API responses to structured records
- Error handling: Individual record and API error handling without stopping entire sync
- Comprehensive logging: Uses Fivetran's logging framework for troubleshooting and monitoring
```
`SDK v2.0.0+ compliance: Uses direct operation calls (no use of `yield` statements)` Is this really a feature? It's **us** talking how **our** example connector is compliant with **our** standards.

- Data preservation: Maintains all original field names and values where possible
- Primary key generation: Uses natural keys from the API (id fields)
- Upsert operations: Delivers data to Fivetran using upsert operations for each record

## Error handling

The connector implements comprehensive error handling strategies in the `update` function and API request helpers:

- Individual record handling: Failed records don't stop the entire sync process
- API error management: HTTP and XML errors are caught and logged with severity levels
- Timeout configuration: 30-second timeout for API requests to prevent hanging
- Detailed logging: Uses Fivetran's logging framework (INFO, WARNING, ERROR levels)

Refer to the `update` and `make_api_request` functions for error handling and record processing logic.

## Tables created

The connector creates the following tables in your destination:

| Table name        | Primary key                              | Description                                 |
|-------------------|------------------------------------------|---------------------------------------------|

> | `comments`        | `[response_id]`                          | Customer feedback comments                  |
> | `blast_results`   | `[blast_id, response_id, thermometer_id]`| Results from feedback collection campaigns  |
> | `recipient_lists` | `[id]`                                   | Lists of feedback recipients                |
> | `thermometers`    | `[id]`                                   | Configured feedback collection tools        |
> | `metrics`         | `[metric_name, recorded_at]`             | Aggregated feedback metrics                 |

Reviewer: ```suggestion
| `COMMENTS`        | `[response_id]`                          | Customer feedback comments                  |
| `BLAST_RESULTS`   | `[blast_id, response_id, thermometer_id]`| Results from feedback collection campaigns  |
| `RECIPIENT_LISTS` | `[id]`                                   | Lists of feedback recipients                |
| `THERMOMETERS`    | `[id]`                                   | Configured feedback collection tools        |
| `METRICS`         | `[metric_name, recorded_at]`             | Aggregated feedback metrics                 |
```
